### DISCUSSION TITLE
Time dependent BCs

### DISCUSSION THREAD

**Discussion by AmbroiseJuston** [idaholab/moose#28229] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello, I have some problems understanding and implementing a Boundary condition that varies with time. I need a  Dirichlet Boundary condition that goes from a to b (for example 0 to 1). I found that MOOSE does have this function in terms of "FunctionDirichletBC" however I need help with the syntax and formatting of the function.

As background I specifically want Voltage (V) to vary with time only and apply it to the top edge of a 2D mesh, going from 0 up to 1 and then back down to 0 linearly. To apply said BC would it be sufficient to add this block:
```
[BCs]
  [./V_top]
    type = FunctionDirichletBC
    variable = V
    boundary = 'top'
    function = - abs(t - 1) + 1
  [../]
[] 
```
Does MOOSE recognise t to be time? I noticed there is also the PiecewiseLinear function that could be added, would I have to instead create a different function block with an input .cvs and then link that to the function input of the FunctionDirichletBC shown above?

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are numerous ways to input these functions, the PiecewiseLinear would work too. I believe 
```
function = '- abs(t - 1) + 1'
```
will already get parsed automatically into a function of time.
If not, then you can use a ParsedFunction `f` and specify `function = f` in that object.
For f you will have
```
[Functions]
  [f]
    type = ParsedFunction
    expression = '- abs(t - 1) + 1'
  []
[]
```

**Reply by AmbroiseJuston** [reply]
Thank you very much, it did indeed seem to work. I would also like to output the voltage into a CVS file, As a way to double check, how would I add that to the input file/what documentation exists that would lead me to implement this?

**Reply by GiudGiud** [reply]
you can output functions to console and csv with this object
https://mooseframework.inl.gov/source/postprocessors/FunctionValuePostprocessor.html
